### PR TITLE
W-13260932: add APIkit for REST & SOAP overviews

### PR DESCRIPTION
**Pull request by juanchib**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by dcourtil — APPROVED** [review]

**Review by iglosiggio — APPROVED** [review]

File: modules/ROOT/pages/apikit-for-soap.adoc [hunk 1]
```diff
@@ -3,4 +3,23 @@ ifndef::env-site,env-github[]
...
+
+* *Automatic binding to WSDL*: APIkit for SOAP automatically binds to your WSDL definitions, saving you time and effort in configuring your APIs.
+
+* *Efficient handling of SOAP messages*: With APIkit for SOAP, you can seamlessly handle SOAP request and response messages, enabling smooth communication between SOAP-based services.
+
+* *Support for data transformations*: The framework provides robust support for data transformations, enabling you to easily manipulate and format data within SOAP APIs.
+
+By placing emphasis on SOAP/WSDL contracts, APIkit for SOAP enables you to define and expose SOAP APIs with ease. Its capabilities in managing SOAP message formats and headers ensure seamless integration with SOAP infrastructure. By leveraging the features of APIkit for SOAP, you can streamline the SOAP API development process, resulting in more efficient and reliable SOAP-based APIs.
+
+With APIkit for SOAP, you can efficiently create robust and reliable SOAP-based APIs, accelerating your development workflow and delivering enhanced integration capabilities.
```
Comment (dcourtil) [inline modules/ROOT/pages/apikit-for-soap.adoc]: All of this sound pretty similar to what is mentioned above, almost with the same words but in different order. In my opinion I would get rid of it. Maybe keep something that was not mentioned or summarize it.
```suggestion
With APIkit for SOAP, you can efficiently implement robust and reliable SOAP-based APIs, accelerating your development workflow and delivering enhanced integration capabilities.
```

File: modules/ROOT/pages/overview-apikit-for-rest.adoc [hunk 2]
```diff
@@ -3,4 +3,25 @@ ifndef::env-site,env-github[]
...
 
-TBD
\ No newline at end of file
+APIkit for REST is a powerful framework designed to simplify the development and integration of RESTful APIs. REST (Representational State Transfer) is an architectural style for building web services, and APIkit for REST provides a comprehensive set of tools and features to streamline REST API development while ensuring compliance with industry standards.
+
+APIkit for REST fully supports RAML (RESTful API Modeling Language) and OAS (OpenAPI Specification), which are widely adopted formats for describing RESTful APIs. By leveraging RAML or OAS, you can define your API contracts, enabling consistent design, enhanced collaboration, and clear documentation.
+
+Features:
+
+* *Automatic Validation*: APIkit for REST automatically validates incoming requests and outgoing responses against your API contracts defined in RAML or OAS. This ensures that your APIs adhere to the specified rules and constraints, enhancing the overall quality and reliability of your API implementation.
```
Comment (dcourtil) [inline modules/ROOT/pages/overview-apikit-for-rest.adoc]: APIkit does not validate outgoing responses for any protocol.
```suggestion
* *Automatic Validation*: APIkit for REST automatically validates incoming requests against your API contracts defined in RAML or OAS. This ensures that requests adhere to the specified rules and constraints of the API, enhancing the overall quality and reliability of your API implementation.
```

File: modules/ROOT/pages/overview-apikit-for-rest.adoc [hunk 3]
```diff
@@ -3,4 +3,25 @@ ifndef::env-site,env-github[]
...
+
+* *Code Generation*: The framework provides code generation capabilities that help streamline the implementation of your RESTful APIs. APIkit for REST can generate implementation templates based on your API contracts, reducing the amount of manual coding required and accelerating the development process.
+
+* *Intelligent Routing*: With APIkit for REST, incoming requests are intelligently routed to the appropriate endpoints based on the defined API contracts. This routing mechanism simplifies request handling, improves performance, and ensures that requests are processed by the correct API resources.
```
Comment (dcourtil) [inline modules/ROOT/pages/overview-apikit-for-rest.adoc]: In addition to the above feature, I think this is an excellent explanation for APIkit in general, not only for REST. I would move it to the general APIkit overview (if not already there).

File: modules/ROOT/pages/overview-apikit-for-rest.adoc [hunk 4]
```diff
@@ -3,4 +3,25 @@ ifndef::env-site,env-github[]
...
+
+APIkit for REST simplifies the development of RESTful APIs by providing a unified and consistent approach. By leveraging industry-standard formats like RAML or OAS, you can define your API contracts, ensuring a clear understanding of the API capabilities and behavior. The automatic validation, code generation, and intelligent routing features further streamline the development process, improving efficiency and reducing implementation effort.
+
+With APIkit for REST, you can build robust and reliable RESTful APIs. The automatic validation feature ensures that requests and responses conform to the defined API contracts, reducing the risk of errors and enhancing the overall quality of your APIs. This increased reliability is crucial for successful API-led implementations and seamless integration with other systems.
+
+APIkit for REST is an indispensable tool for efficient REST API development and integration. By leveraging its features and adhering to industry standards, you can accelerate your API development lifecycle, improve collaboration, and deliver high-quality RESTful APIs.
```
Comment (dcourtil) [inline modules/ROOT/pages/overview-apikit-for-rest.adoc]: Also, this sounds repeated story but with different words.
```suggestion

```

**Review by dcourtil — CHANGES_REQUESTED** [review]
Please, let me know what do you think

File: modules/ROOT/pages/overview-apikit-for-rest.adoc (hunk 2 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/overview-apikit-for-rest.adoc]: Thanks!

File: modules/ROOT/pages/overview-apikit-for-rest.adoc (hunk 3 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/overview-apikit-for-rest.adoc]: Added to the APIkit overview. Thanks!

File: modules/ROOT/pages/overview-apikit-for-rest.adoc (hunk 4 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/overview-apikit-for-rest.adoc]: Let me go over it and avoid repetition.
Thanks for mentioning it!

File: modules/ROOT/pages/apikit-for-soap.adoc (hunk 1 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/apikit-for-soap.adoc]: Thanks!

File: modules/ROOT/pages/apikit-4-forgraphql.adoc [hunk 5]
```diff
@@ -1,17 +0,0 @@
...
-GraphQL is a data query and manipulation language that enables API consumers to ask for data that is relevant only to them. The APIkit for GraphQL module works with Anypoint Code Builder, enabling you to develop Mule applications. This module supports implementing GraphQL services, which use connectors to map GraphQL APIs to the existing backends where source data is stored.
-
-The APIkit scaffolder enables you to quickly implement projects by generating the initial skeleton of the application at design time based on the GraphQL schema. The application design maps queries, mutations, and types in your API to the Mule flows so you only specify how your data sources fulfill a particular data type.
-
-At run time, the APIkit for GraphQL router analyzes the query graph and orchestrates the execution of Mule flows based on the requests. The result of each flow execution is combined to return a result that mirrors the shape of the requested query.
-
-== See Also
-
-* xref:anypoint-code-builder::implement-a-graphql-api.adoc[]
-* xref:install-graphql-module.adoc[]
```
Comment (iglosiggio) [inline modules/ROOT/pages/apikit-4-forgraphql.adoc:17]: Is this removal OK?

File: modules/ROOT/pages/apikit-for-soap.adoc [hunk 6]
```diff
@@ -1,6 +0,0 @@
-= APIkit for SOAP
-ifndef::env-site,env-github[]
-include::_attributes.adoc[]
-endif::[]
-
-TBD
```
Comment (iglosiggio) [inline modules/ROOT/pages/apikit-for-soap.adoc:6]: Is this removal OK?

File: modules/ROOT/pages/overview-apikit-for-rest.adoc [hunk 7]
```diff
@@ -1,6 +0,0 @@
-= APIkit for REST
-ifndef::env-site,env-github[]
-include::_attributes.adoc[]
-endif::[]
-
-TBD
```
Comment (iglosiggio) [inline modules/ROOT/pages/overview-apikit-for-rest.adoc:6]: Is this removal OK?

File: modules/ROOT/pages/apikit-4-forgraphql.adoc (hunk 5 above)
Comment (iglosiggio) [reply modules/ROOT/pages/apikit-4-forgraphql.adoc:17]: Ahh, this is basically `modules/ROOT/pages/apikit-4-for-graphql.adoc`

File: modules/ROOT/pages/apikit-for-soap.adoc (hunk 6 above)
Comment (iglosiggio) [reply modules/ROOT/pages/apikit-for-soap.adoc:6]: Ahh, this is basically `modules/ROOT/pages/apikit-4-for-soap.adoc`

File: modules/ROOT/pages/overview-apikit-for-rest.adoc (hunk 7 above)
Comment (iglosiggio) [reply modules/ROOT/pages/overview-apikit-for-rest.adoc:6]: Ahh, this is basically `modules/ROOT/pages/apikit-4-for-rest.adoc`

File: modules/ROOT/pages/apikit-4-for-soap.adoc [hunk 8]
```diff
@@ -1,20 +1,19 @@
...
+* Efficient handling of SOAP messages
+* Support for data transformations
+
+By placing emphasis on SOAP/WSDL contracts, APIkit for SOAP enables you to define and expose SOAP APIs with ease. Its capabilities in managing SOAP message formats and headers ensure seamless integration with SOAP infrastructure. With APIkit for SOAP, you can efficiently create robust and reliable SOAP-based APIs, accelerating your development workflow and delivering enhanced integration capabilities.
 
 == See Also
 
-* xref:configure-cloudhub-path-task.adoc[]
-* xref:apikit-4-xml-reference.adoc[]
+* xref:apikit-4-soap-prerequisites-task.adoc[]
```
Comment (iglosiggio) [inline modules/ROOT/pages/apikit-4-for-soap.adoc:18]: I would link to `implementing-apikit-4-for-soap.adoc` instead

**Review by dcourtil — APPROVED** [review]

**Pull request merged**